### PR TITLE
#165413104 Enable Social Login via Google, Facebook and twitter

### PR DESCRIPTION
**Pull request by Niyitangasam**
Description
=======
This PR implements login functionalities with pre-existing social media accounts. Login strategies for facebook, twitter and google are implemented. It also adds the functionality of keeping users info in case they will use the system next time

Type of change
=======
N/A

How has it been tested
=======
It can be tested through below links

- ```api/v1/login/google```
- ```api/v1/login/facebook```
- ```api/v1/login/twitter```

Checklist:
=======
N/A

Pivotal tracker story ID
=======
[165413099](https://www.pivotaltracker.com/n/projects/2326984/stories/165413104)

### CODE REVIEW

File: controllers/user.controller.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import db from '../models/index';
```
[the remark this answers is not in the document]
Comment (AJAkimana) [reply controllers/user.controller.js]: User `import { User } from '../models' ` instead of `import db from '../models/index'`, for more readable code

**Review by AJAkimana — COMMENTED** [review]
Good work man, but I would suggest using sequelize functions instead to create the other function that requires sequelize functions. For instance the function `findRecord` I think it is not needed.

File: controllers/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (macphilips) [inline controllers/index.js]: Kindly remove this line since there is no significant change in this file

**Review by jabichris — APPROVED** [review]

**Review by Kabalisa — APPROVED** [review]

**Review by Kabalisa — APPROVED** [review]

**Review by AJAkimana — APPROVED** [review]

File: helpers/passport.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+
+const verifyCallback = async (accessToken, refreshToken, profile, done) => {
+  let user;
+  try {
+    console.log(accessToken);
```
Comment (macphilips) [inline helpers/passport.js]: remove `console.log`

File: package.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
-    "test": "nyc mocha --require @babel/register \"tests/**/*.js\"",
+    "test": "NODE_ENV=test nyc mocha --require @babel/register --require @babel/polyfill \"tests/**/*.js\"",
     "lint": "eslint --ignore-path .gitignore .",
     "lint:fix": "npm run lint -- --fix",
     "start:dev": "nodemon --exec babel-node index.js",
```
Comment (macphilips) [inline package.json]:
```suggestion
    "start:dev": "nodemon --exec node --require @babel/register --require dotenv/config index.js",
```

**Review by macphilips — CHANGES_REQUESTED** [review]

File: config/passport/index.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+/* eslint-disable require-jsdoc */
```
Comment (macphilips) [inline config/passport/index.js]: Remove this comment

**Review by macphilips — CHANGES_REQUESTED** [review]

File: helpers/index.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable require-jsdoc */
```
Comment (macphilips) [inline helpers/index.js]: Remove comment

**Pull request merged**